src/ffi.rs

```diff
@@ -948,3 +1001,57 @@ pub extern fn quiche_conn_peer_streams_left_bidi(conn: &mut Connection) -> u64 {
 pub extern fn quiche_conn_peer_streams_left_uni(conn: &mut Connection) -> u64 {
     conn.peer_streams_left_uni()
 }
+
+fn std_addr_from_c(addr: &sockaddr, addr_len: socklen_t) -> SocketAddr {
+    unsafe {
+        match addr.sa_family as i32 {
+            libc::AF_INET => {
+                assert!(addr_len as usize == std::mem::size_of::<sockaddr_in>());
+
+                SocketAddr::V4(
+                    *(addr as *const _ as *const sockaddr_in as *const _),
+                )
+            },
+
+            libc::AF_INET6 => {
+                assert!(addr_len as usize == std::mem::size_of::<sockaddr_in6>());
+
+                SocketAddr::V6(
+                    *(addr as *const _ as *const sockaddr_in6 as *const _),
+                )
+            },
+
+            _ => unimplemented!("unsupport address type"),
+        }
+    }
+}
+
+fn std_addr_to_c(addr: &SocketAddr) -> (sockaddr_storage, socklen_t) {
```

Comment:
seems to be a bit of repition here, could I guess you need the destructuring to get to the actual `addr`, so any suggestion came up with doesn't really help much.

Reply:
Yeah, it might be possible to make a macro or something, but every time I try to Rust macros I get confused, so I haven't tried that yet.